modules/foreman_host.py

```diff
@@ -0,0 +1,172 @@
+requirements:
+    - "nailgun >= 0.28.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        required: false
+    hostname:
+        description:
+            - Hostname
+        required: true
+    mac:
+        description:
+            - MAC HOST
+        required: true
+    organization:
+        description:
+            - Organization
+        required: true
+    location:
+        description:
+            - Location
+        required: true
+    architecture
+        description:
+            - Architecture
+        required: true
+    operatingsystem
+        description:
+            - Operating System
+        required: true
+'''
+
+EXAMPLES = '''
+- name: "Create Host"
+  foreman_host:
+    username: "admin"
+    password: "admin"
+    server_url: "https://foreman.example.com"
+    hostname: "hostname"
+    mac: "00:00:00:00:00:00"
+    organization: "Default Organization"
+    location: "Default Location"
+    architecture: "x86_64"
+    operatingsystem: "Fedora 26"
+  delegate_to: localhost
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+            Host,
+    )
+    from ansible.module_utils.ansible_nailgun_cement import (
+            create_server,
+            create_entity,
+            update_entity,
+            handle_no_nailgun,
+            find_organization,
+            find_operatingsystem,
+            find_architecture,
+            find_location,
+            ping_server,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+def create_host(module, hostname, mac, organization, location, architecture, operatingsystem):
+    changed = False
+    location_id = find_location(module, location)
+    organization_id = find_organization(module, organization)
+    architecture_id = find_architecture(module, architecture)
+    operatingsystem_id = find_operatingsystem(module, operatingsystem)
+
+    host = Host(name=hostname)
+    host = host.search(set(), {'search': 'name="{}"'.format(hostname)})
+
+    if len(host) == 0:
+        props = { 'name': hostname, 'organization': organization_id, 'location': location_id, 'mac': mac, 'architecture': architecture_id, 'operatingsystem': operatingsystem_id }
+        create_entity(Host, props, module)
+        changed = True
+    else:
+        props = { 'name': hostname, 'organization': organization_id, 'location': location_id, 'mac': mac, 'architecture': architecture_id, 'operatingsystem': operatingsystem_id }
+        update_entity(host[0], props, module)
+        changed = True
+
+    return changed
```

Comment:
In this code path, `changed` will always end up being `True`, yet slightly obfuscated.
Since a host can be created, updated or even deleted, consider using `naildown_entity_state` together with a parameter `state`. Also `state` can be used, to trigger power states of the host.